coap-gateway/service/resourceDiscovery.go

```diff
@@ -62,7 +60,7 @@ func makeDiscoveryResp(isTLSListener bool, serverAddr string, getResourceLinksCl
 			break
 		}
 		if err != nil {
-			return nil, status.Convert(err).Code(), fmt.Errorf("cannot create discovery response: %w", err)
+			return nil, fmt.Errorf("cannot create discovery response: %w", err)
```

Comment:
Does `GrpcErr2CoapCode` still work correctly when you wrap the original `err` with `fmt.Errorf`?

Reply:
Yes. It uses https://pkg.go.dev/errors#As which unwraps grpc error. If it is not a grpc error then `GrpcErr2CoapCode ` returns an internal error. (`codes.Unknown -> coapCodes.InternalServerError`)